articles/migrations/guides/migration-oauthro-oauthtoken-pwdless.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Migration Guide for Resource Owner Passwordless Authentication
+description: Learn how to migrate your Passwordless API calls and responses from /oauth/ro to /oauth/token
+toc: true
+contentType:
+  - concept
+useCase:
+  - secure-an-api
+  - migrate
+---
+# Migration Guide for Resource Owner Passwordless Credentials Exchange
+
+Support for Resource Owner Password was added to [/oauth/token](/api/authentication#authorization-code). Usage of the [/oauth/ro](/api/authentication#resource-owner) endpoint was deprecated on July 08, 2017. This endpoint was used to exchange an OTP received by the end-user by email or SMS with for an `id_token` and an `access_token`.
+
+## Does this affect me ?
+
+This guide is for users who use the resource owner passwordless credentials exchange, and call `/oauth/ro` directly, without the use of any Auth0 libraries or SDKs. 
+
+## Changes to requests
+
+Previously, the payload of a request to `/oauth/ro` looked similar to this:
+
+```json
+{
+  "grant_type": "password",
+  "client_id": "123",
+  "username": "alice",
+  "password": "A3ddj3w", 
+  "connection": "my-database-connection",
+  "scope": "openid email favorite_color offline_access",
+  "device": "my-device-name"
+}
+```
+
+* The endpoint to execute token exchanges is now `/oauth/token`
+* [Auth0's own grant type](/api-auth/tutorials/password-grant#realm-support) is used to authenticate users from a specific connection (or `realm`). 
+* Auth0 supports the [standard OIDC scopes](/scopes/current/oidc-scopes), along with the <dfn data-key="scope">scopes</dfn> which you have defined in your [custom API](/api-auth/apis).
+* A scope that does not fit in one of these categories, such as the above `favorite_color`, is no longer a valid scope.
+* The `device` parameter is removed.
+* The <dfn data-key="audience">`audience`</dfn> parameter is optional.
+
+Here is an example of the payload of a request to `/oauth/token`:
+
+```json
+{
+  "grant_type" : "http://auth0.com/oauth/grant-type/passwordless/otp",
+  "client_id": "YOUR_CLIENT_ID",
+  "client_secret": "YOUR_CLIENT_SECRET", // only for web apps, native apps don’t have a client secret
+  "username": "<email address>", // or "<phone number>"
+  "otp": "CODE",
+  "realm": "email", // or "sms" 
+  "audience" : "your-api-audience", // in case you need an access token for a specific API
+  "scopes": "openid profile email" // whatever scopes you need
+}
+```
+
+* The grant type is specified here as `http://auth0.com/oauth/grant-type/passwordless/otp`
```

Comment:
Should this (here and in the example above) be `https`?

Reply:
No, the grants are actually a 'string' more than an URL, and we have been using http, e.g. https://auth0.com/docs/api-auth/tutorials/password-grant#realm-support